src/app/modules/trezor/trezorAccount.js

```diff
@@ -0,0 +1,66 @@
+import * as nemSdk from "nem-sdk";
+import { Observable } from "rxjs";
+import { PublicAccount, NEMLibrary } from "nem-library";
+
+/**
+ * TrezorAccount model
+ */
+export class TrezorAccount {
+ 
+    /**
+     * Constructor
+     * @param address
+     */
+    constructor(address, hdKeyPath, Trezor) {
+        this.address = address;
+        this.hdKeyPath = hdKeyPath;
+        this._Trezor = Trezor;
+    }
+
+    /**
+     * Sign a transaction
+     * @param transaction
+     * @param receiverPublicKey (optional): if given, the message will be encrypted
+     * @returns {{data: any, signature: string}}
+     */
+    signTransaction(transaction, receiverPublicKey) {
+        transaction.signer = PublicAccount.createWithPublicKey("462ee976890916e54fa825d26bdd0235f5eb5b6a143c199ab0ae5ee9328e08ce");
+        transaction.setNetworkType(NEMLibrary.getNetworkType());
+        const dto = transaction.toDTO();
+        if (receiverPublicKey !== undefined) {
+            dto.message.type = 2;
+            dto.message.publicKey = receiverPublicKey;
+
+            const encPayload =
+            nemSdk.default.crypto.helpers.encode(receiverPublicKey, receiverPublicKey, dto.message.payload);
+
+            let min = Math.floor(Math.max(1, (dto.amount / 1000000) / 10000));
+            min = min > 25 ? 25 : min;
+            let fee = Math.floor(0.05 * min * 1000000);
+            if (dto.message.payload && encPayload.length != 0) {
+                fee += 0.05 * (Math.floor((encPayload.length / 2) / 32) + 1) * 1000000;
+            }
```

Comment:
@cryptoBeliever I'm keen to refactor here a little bit, extract some of the values to constants but not sure how to name them, any ideas on what the following values are? => 10000, 25, 0.05

Reply:
Magic numbers :) As I can see `receiverPublicKey` is never passed to this method. So maybe just remove it and all inside if?

Reply:
Removed & updated.

Reply:
it is part of the transfer transaction fee calculation. it should match this code:
https://github.com/NemProject/nem.core/blob/dev/src/main/java/org/nem/core/model/AbstractTransactionFeeCalculator.java#L32